osu.Game/OsuGame.cs

```diff
@@ -215,31 +215,102 @@ private void load(FrameworkConfigManager frameworkConfig)
 
         private ExternalLinkOpener externalLinkOpener;
 
-        public void OpenUrlExternally(string url)
+        /// <summary>
+        /// Handle an arbitrary URL. Displays via in-game overlays where possible.
+        /// This can be called from a non-thread-safe non-game-loaded state.
+        /// </summary>
+        /// <param name="url">The URL to load.</param>
+        public void HandleLink(string url) => Schedule(() => HandleLink(MessageFormatter.GetLinkDetails(url)));
```

Comment:
The schedule should exist on the other method. This also means that the other method isn't actually thread-safe.

Reply:
Fair call